tests/testthat/test-scatter_plot_financial.R

```diff
@@ -1,36 +1,97 @@
 test_that("returns an object of the expected class", {
-  p <- scatter_plot_financial(financial, benchmarks())
+  data <- example_financial(
+    !!aka("scenario") := "IPR",
+    !!aka("year") := 1,
+    !!aka("tilt_sector") := "t"
+  )
+  p <- scatter_plot_financial(data, benchmarks(), scenario = "IPR", year = 1)
   expect_s3_class(p, "ggplot")
 })
 
 test_that("returns correct risk categories values", {
-  p <- scatter_plot_financial(financial, benchmarks())
+  data <- example_financial(
+    !!aka("scenario") := "IPR",
+    !!aka("year") := 1,
+    !!aka("tilt_sector") := "t"
+  )
+  p <- scatter_plot_financial(data, benchmarks(), scenario = "IPR", year = 1)
   risk_names <- unique(p$data$emission_profile)
   possible_names <- c("low", "medium", "high", "other")
   expect_true(all(risk_names %in% possible_names))
 })
 
 test_that("returns correct benchmarks values", {
-  plot <- scatter_plot_financial(financial, benchmarks())
-  benchmarks <- unique(plot$data$benchmark)
-  expected_benchmarks <- financial |>
+  data <- example_financial(
+    !!aka("scenario") := "IPR",
+    !!aka("year") := 1,
+    !!aka("tilt_sector") := "t"
+  )
+  p <- scatter_plot_financial(data, benchmarks(), scenario = "IPR", year = 1)
+  benchmarks <- unique(p$data$benchmark)
+  expected_benchmarks <- data |>
     pull(benchmark) |>
     unique()
   expect_true(all(benchmarks %in% expected_benchmarks))
 })
 
 test_that("calculate_rank handles NAs for any mode", {
-  data <- tibble(
-    bank_id = c("a", "b"),
-    profile_ranking = c(1, NA),
-    equal_weight_finance = c(1, 1),
-    best_case_finance = c(1, 1),
-    worst_case_finance = c(1, 1)
+  data <- example_financial(
+    !!aka("scenario") := "IPR",
+    !!aka("year") := 1,
+    !!aka("tilt_sector") := "t",
+    !!aka("profile_ranking") := NA
   )
+
   results <- list(
     calculate_rank(data, "equal_weight_finance", aka("profile_ranking")),
     calculate_rank(data, "best_case_finance", aka("profile_ranking")),
     calculate_rank(data, "worst_case_finance", aka("profile_ranking"))
   )
   expect_true(all(unlist(lapply(results, function(x) all(x$rank == 1)))))
 })
+
+test_that("calculated ranks are between 0 and 1 for each scores", {
+  test_rank <- function(data, mode, column) {
+    result <- calculate_rank(data, mode, column)
+    rank <- result[[1]]
+    expect_true(all(rank >= 0 & rank <= 1))
+  }
+
+  data <- example_financial(
+    !!aka("scenario") := "IPR",
+    !!aka("year") := 1,
+    !!aka("tilt_sector") := "t"
+  )
+
+  lapply(modes(), function(mode) {
+    test_rank(data, mode, aka("profile_ranking"))
+  })
+  lapply(modes(), function(mode) {
+    test_rank(data, mode, aka("emission_profile"))
+  })
+})
```

Comment:
This allows you to run many tests with little code. But see my comments above. In tests sometimes it's best to focus on fewer, more meaningful tests than lots' of overlapping ones.